### DISCUSSION TITLE
How do you end interactions with scenes when using widgets?

### DISCUSSION THREAD

**Discussion by adam-grant-hendry** [pyvista/pyvista#3523] [Q&A]
The docs state in a number of places that

> After interacting with the scene, the clipped mesh is available as...

but it doesn't say how you end interaction with the scene? Is it a key press? Closing the window? Can a key press, like `'q'`, be used to end the interaction?

**Comment by adam-grant-hendry** [comment]
The `pyvista` [source ](https://github.com/pyvista/pyvista/blob/068e4922742060bfe4f524955edb4ec7f4ee9782/pyvista/plotting/widgets.py) shows it's simply a VTK `Observer` tied to the specified callback with a VTK `EndInteractionEvent`, so a `KeyPressEvent` should work here if passed in as `interaction_event`:

https://github.com/pyvista/pyvista/blob/068e4922742060bfe4f524955edb4ec7f4ee9782/pyvista/plotting/widgets.py#L43-L56

https://github.com/pyvista/pyvista/blob/068e4922742060bfe4f524955edb4ec7f4ee9782/pyvista/plotting/widgets.py#L147-L153

**Comment by banesullivan** [comment]
Sorry for the slow response.

> Is it a key press? Closing the window? Can a key press, like 'q', be used to end the interaction?

Closing the window is what we mean by "After interacting with the scene". this would be hitting the `q`-key or the exit button in the window. 

If using `pyvistaqt`, the data would be immediately available on the plotter.

Hope this helps!
